arch/arm/core/aarch64/switch.S

```diff
@@ -222,6 +220,7 @@ offload:
 	b	inv
 
 context_switch:
+	ldr	x17, [x17]
```

Comment:
Another SMP note: You're going to want to store through that switched_from pointer at the end of all context saving with some non-null value (and on ARM include a write barrier, IIRC).  See the comments (they're relatively new) at https://github.com/zephyrproject-rtos/zephyr/blob/master/kernel/include/kernel_arch_interface.h#L77

Basically this is used as a synchronization mechanism to be sure the thread being saved is in a valid state.